compose: merge latest spec changes (#15539)
docker/docker.github.io: compose/compose-file/index.md

If set to `true`, `external` specifies that this network’s lifecycle is maintained outside of that of the application.
Compose Implementations SHOULD NOT attempt to create these networks, and raises an error if one doesn't exist.

> If `external` is set to `true` and network configuration has other but `name` attributes set, considering resource is
> not managed by compose lifecycle, Compose Implementations SHOULD reject a Compose file as invalid.

Reviewer: I'm not sure about this passage:
```suggestion
If `external` is set to `true` and network configuration has other attributes set besides `name` , resource consideration is not managed by the Compose lifecycle. In these cases, Compose Implementations SHOULD reject the Compose file as invalid.
```

Author: I believe what it's saying is that for anything `external: true`, the only other allowable attribute is `name`. If other attributes are defined, Compose Implementations SHOULD reject the Compose file as invalid.

Perhaps:
```
If `external` is set to `true`, the resource is not managed by Compose. If `external` is set to `true` and the network configuration has other attributes set besides `name` , Compose Implementations SHOULD reject the Compose file as invalid.
```

of that of the application. Compose implementations MUST NOT attempt to create these volumes, and MUST return an error if they
do not exist.

> If `external` is set to `true` and volume configuration has other but `name` attributes set, considering resource is
> not managed by compose lifecycle, Compose Implementations SHOULD reject a Compose file as invalid.

Reviewer: ```suggestion
If `external` is set to `true` and volume configuration has other but `name` attributes set, resource consideration is
not managed by the Compose lifecycle. In these cases, Compose Implementations SHOULD reject the Compose file as invalid.
```

    name: "${HTTP_CONFIG_KEY}"
```

> If `external` is set to `true` and secret configuration has other but `name` attributes set, considering resource is
> not managed by compose lifecycle, Compose Implementations SHOULD reject a Compose file as invalid.

Reviewer: ```suggestion
If `external` is set to `true` and secret configuration has other attributes set besides `name` , resource consideration is
not managed by the Compose lifecycle. In these cases, Compose Implementations SHOULD reject the Compose file as invalid.
```
